tests/10apidoc/33room-members.pl

```diff
@@ -324,6 +324,13 @@ sub matrix_create_and_join_room
 
       log_if_fail "room_id=$room_id";
 
+      Future->needs_all(
+        ( map {
+            my $user = $_;
+            matrix_invite_user_to_room( $creator, $user, $room_id );
+         } @other_members)
+      );
```

Comment:
Yeah; I'd do something like

    ( $options{with_invite} ?
       _invite_users( $creator, $room_id, @other_members ) : 
       Future->done() )

with that defined as a helper function above/below.